framework/src/build-task-list.ts

```diff
@@ -5,14 +5,16 @@ import {
 export default (tasks: Task[], blacklist: string[], total: number,) => {
   const internalTasks = [];
   for (const task of tasks) {
-    if (task.pre) {
-      task.pre = task.pre.filter((entry,) => ! blacklist.includes(entry,),);
-    }
-    if (task.post) {
-      task.post = task.post.filter((entry,) => ! blacklist.includes(entry,),);
-    }
-    for (let i=0; i<total; i ++) {
-      internalTasks.push(task,);
+    if (task.main) {
```

Comment:
`task.main` will always be set, but this is where you could do the filtering by checking `task.id`

Reply:
Should it look something like this? 
```
if (taskId && task.id !== taskId) {
      continue;
    }
```

Reply:
no continue, otherwise looks good. Feel free to extract it into a small function